auth/oidc.go

```diff
@@ -209,76 +210,168 @@ func (op *OIDCProvider) DiscoverConfig() (config *oidc.ProviderConfig, shouldSyn
 		}
 	}
 
-	return config, shouldSync, err
+	return config, ctx, shouldSync, err
 }
 
-func (op *OIDCProvider) FetchCustomProviderConfig(discoveryURL string) (*oidc.ProviderConfig, error) {
+func GetOIDCUsername(provider *OIDCProvider, subject string) string {
+	return fmt.Sprintf("%s_%s", provider.UserPrefix, url.QueryEscape(subject))
+}
 
-	var customConfig OidcProviderConfiguration
+type jsonTime time.Time
+
+type idToken struct {
+	Issuer       string                 `json:"iss"`
+	Subject      string                 `json:"sub"`
+	Audience     audience               `json:"aud"`
+	Expiry       jsonTime               `json:"exp"`
+	IssuedAt     jsonTime               `json:"iat"`
+	NotBefore    *jsonTime              `json:"nbf"`
+	Nonce        string                 `json:"nonce"`
+	AtHash       string                 `json:"at_hash"`
+	ClaimNames   map[string]string      `json:"_claim_names"`
+	ClaimSources map[string]claimSource `json:"_claim_sources"`
+}
 
-	// If discovery URL is empty, use the standard discovery URL
-	if discoveryURL == "" {
-		discoveryURL = strings.TrimSuffix(op.Issuer, "/") + discoveryConfigPath
+type claimSource struct {
+	Endpoint    string `json:"endpoint"`
+	AccessToken string `json:"access_token"`
+}
+
+type audience []string
+
+func GetIDToken(rawIDToken string) (*oidc.IDToken, error) {
```

Comment:
Looking further, it seems like this functionality is already implemented in the square/go-jose.v2 library we're importing elsewhere (ParseSigned, UnsafeClaimsWithoutVerification).  Reusing the existing library instead of reimplementing seems like a better approach.